### ISSUE TITLE
Put method with Content on Linux is very slow

### ISSUE THREAD

**Issue by PavelStefanov** [dotnet/corefx#27199]
.Net core - Runtime v2.0.5
``` csharp
using System;
using System.Diagnostics;
using System.Net.Http;
using System.Text;

namespace Test
{
    class Program
    {

        static void Main(string[] args)
        {
            var httpClient = new HttpClient();            

            // PUT
            using (var request = new HttpRequestMessage(HttpMethod.Put, "http://httpbin.org/put"))
            {
                request.Content = new StringContent("{\"test\": 123 }", Encoding.UTF8, "application/json");

                var stopWatch = new Stopwatch();
                stopWatch.Start();
                using (var response = httpClient.SendAsync(request).Result)
                {
                    stopWatch.Stop();
                    Console.WriteLine("PUT with content - " + stopWatch.Elapsed);
                }
            }            

            using (var request = new HttpRequestMessage(HttpMethod.Put, "http://httpbin.org/put"))
            {
                var stopWatch = new Stopwatch();
                stopWatch.Start();
                using (var response = httpClient.SendAsync(request).Result)
                {
                    stopWatch.Stop();
                    Console.WriteLine("PUT without content - " + stopWatch.Elapsed);
                }
            }            

            // POST
            using (var request = new HttpRequestMessage(HttpMethod.Post, "http://httpbin.org/post"))
            {
                request.Content = new StringContent("{\"test\": 123 }", Encoding.UTF8, "application/json");
                
                var stopWatch = new Stopwatch();
                stopWatch.Start();
                using (var response = httpClient.SendAsync(request).Result)
                {
                    stopWatch.Stop();
                    Console.WriteLine("POST with content - " + stopWatch.Elapsed);
                }
            }            

            using (var request = new HttpRequestMessage(HttpMethod.Post, "http://httpbin.org/post"))
            {
                var stopWatch = new Stopwatch();
                stopWatch.Start();
                using (var response = httpClient.SendAsync(request).Result)
                {
                    stopWatch.Stop();
                    Console.WriteLine("POST without content - " + stopWatch.Elapsed);
                }
            }
        }
    }
}
```

Result:
### Ubuntu 16.04.3 LTS
```
PUT with content - 00:00:02.5108530
PUT without content - 00:00:00.1544149
POST with content - 00:00:00.1640221
POST without content - 00:00:00.1710089
```

### Windows 10 Pro
```
PUT with content - 00:00:00.4485657
PUT without content - 00:00:00.2165181
POST with content - 00:00:00.1581848
POST without content - 00:00:00.1617438
```

Why is Put method with content on Linux so slow? I tried use other content types(Steam/Bytes) and they all works slowly also

**Comment by stephentoub** [comment]
libcurl is choosing to use Expect: 100-continue, and it's delaying sending the content until it hears from the server.  Try setting:
```C#
request.Headers.ExpectContinue = false;
```
and see if that makes an impact.  I just tried on my Ubuntu 17.10 VM:

Before:
```
stoub@SToubUbuntu17102:~/tmp$ dotnet run -c Release
PUT with content - 00:00:02.0555575
PUT without content - 00:00:00.0088871
POST with content - 00:00:00.0099487
POST without content - 00:00:00.0076235
```
After:
```
stoub@SToubUbuntu17102:~/tmp$ dotnet run -c Release
PUT with content - 00:00:00.0508100
PUT without content - 00:00:00.0087921
POST with content - 00:00:00.0412229
POST without content - 00:00:00.0070573
```

**Comment by PavelStefanov** [comment]
@stephentoub thank you. But why does Put send Expect header by default and Post does not?

**Comment by stephentoub** [comment]
I'd need to debug through libcurl to see why it's deciding to do that; in .NET Core 2.0, by default CurlHandler leaves that decision up to libcurl.  In .NET Core 2.1, we suppress Expect: 100-continue unless the developer explicitly opts in to it.

**Comment by Jlalond** [comment]
Hey Guys. I'm going to dig into this and see if I can figure out why this is happening and then write some tests for it

**Comment by karelz** [comment]
Looks like nothing actionable on .NET Core side, closing.

**Comment by PavelStefanov** [comment]
@karelz I don't understand. Will this be fixed in 2.1?

**Comment by karelz** [comment]
@PavelStefanov what kind of fix would you expect in 2.1 beside "_In .NET Core 2.1, we suppress Expect: 100-continue unless the developer explicitly opts in to it._"?

**Comment by PavelStefanov** [comment]
@karelz ok, this is what I expect, thanks!
